### PR TITLE
CLI: Update init script for react-native v6.5

### PR DESCRIPTION
**Pull request by dannyhw**
Issue: #

## What I did

Updates the init script for react native to work with the upcoming v6.5 version.

- I renamed copyComponents to copyTemplates since its more generic
  - I also changed the options to an object since I added a 4th option and it was a lot
  - for RN we copy more than just the components since it has a 2 more configuration files to copy
- removed the react-native template files so they can be copied from the react-native package instead
- added an option to not copy the common files since they don't apply to react-native

## How to test

run init command on a react-native project
If you want to test this then let me know and I'll show you how.

### CODE REVIEW

File: code/lib/cli/src/helpers.ts [hunk 1]
```diff
@@ -186,17 +186,25 @@ export function copyTemplate(templateRoot: string, destination = '.') {
   fse.copySync(templateDir, destination, { overwrite: true });
 }
 
-export async function copyComponents(
-  renderer: SupportedFrameworks | SupportedRenderers,
-  language: SupportedLanguage,
-  destination?: string
-) {
+type CopyComponentsOptions = {
```
Comment (shilman) [inline code/lib/cli/src/helpers.ts:189]:
```suggestion
type CopyTemplateFilesOptions = {
```

**Review by shilman — APPROVED** [review]
Changes look good -- one small nit!

Would be great if @kasperpeulen could also check

File: code/lib/cli/src/helpers.ts (hunk 1 above)
Comment (dannyhw) [reply code/lib/cli/src/helpers.ts]: Nice catch! Updated it.

**Review by kasperpeulen — APPROVED** [review]
Looks great!

**Pull request merged**